src/libopensc/card-idprime.c

```diff
@@ -111,6 +203,23 @@ idprime_private_data_t *idprime_new_private_data(void)
 		return NULL;
 	}
 
+	/* Initialize container list */
+	if (list_init(&priv->containers) != 0 ||
+	    list_attributes_comparator(&priv->containers, list_comparator_uint8_t) != 0 ||
```

Comment:
I think you still need a comparator as you need to compare the member `index`, not the whole `idprime_container_t` casted to `uint8_t`. Same for the keyrefmap